src/space/objects/realization/loop-hole.js

```diff
@@ -0,0 +1,123 @@
+import Vector from "../../utils/vector";
+import EncounterObject from "../pattern/encounterObject";
+import Type from "../pattern/encounterObjectType";
+import MyMath from "../../utils/myMath";
+import AssetLoader from "../../utils/assetLoader";
+
+const LH_SPRITE_RADIUS = 20;
+const LH_MIN_DIS = 700;
+//LH_ENTRY_SOURCE_Y is the y coordinate of position in loophole png file
+const LH_ENTRY_SOURCE_Y = 40;
+//LH_EXIT_SOURCE_Y is the y coordinate of position in loophole png file
+const LH_EXIT_SOURCE_Y = 120;
+// the change value of globalAlpha each time
+const LH_FADE_DEGREE = 5;
+
+
+export default class LoopHole extends EncounterObject{
+
+    /**
+      * Construction for a new loop hole
+      *
+      * @param {Encounter} encounter
+      * @param {Vector} position
+      */
+    constructor(encounter, position){
+        super(encounter, Type.LOOP_HOLE, LH_SPRITE_RADIUS, position);
+	    this.mapWidth = encounter.width;
+	    this.mapHeight = encounter.height;
+        // generate loop hole exit
+      	do{
+      		this.outX = Math.floor(Math.random()*this.mapWidth);
+      		this.outY = Math.floor(Math.random()*this.mapHeight);
+            this.outPos = new Vector(this.outX, this.outY);
+      	} while (MyMath.distance(position, this.outPos) <= LH_MIN_DIS);
+        //load image
+    	this.lhImage = AssetLoader.getAsset("loophole");
```

Comment:
I think I have mentioned it a few times already, but starting with this line and above your indentation is off. Please, look into it, fix it and check in Github UI, that it is ok.